src/main/java/duke/Ui.java

```diff
@@ -0,0 +1,124 @@
+package duke;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Ui {
+
+    public Ui() {
+        textWarper("Hello! I'm Duke \nWhat can I do for you?");
+    }
+
+    public void showBye() {
+        textWarper("Bye. Hope to see you again soon!\n");
+    }
+
+    /**
+     * Prints out the TaskList in a nice format.
+     * 
+     * @param tasks the TaskList to be printed.
+     */
+    public void showList(TaskList tasks) {
+        String out = "Here are the tasks in your list: \n";
+        out += printList(tasks);
+        textWarper(out);
+    }
+
+    /**
+     * Searches the TaskList and prints out the items that matches the user
+     * specifications.
+     * 
+     * @param tasks  the TaskList to be searched.
+     * @param toFind the String that the user is looking for.
+     */
+    public void showFind(TaskList tasks, String toFind) {
+        String out;
+        List<Task> tempList = new ArrayList<>();
+        for (Task t : tasks.getTasks()) {
+            if (t.getName().contains(toFind)) {
+                tempList.add(t);
+            }
+        }
+        if (!tempList.isEmpty()) {
+            out = "Here are the matching tasks in your list: \n";
+            out += printList(new TaskList(tempList));
+        } else {
+            out = "Nothing in the list matches\n";
+        }
+        textWarper(out);
+    }
+
+    /**
+     * Prints out confirmation after marking an item as done.
+     * 
+     * @param input The string input by the user.
+     * @param tasks The TaskList to be checked.
+     */
+    public void showDone(String input, TaskList tasks) {
+        String out = "Nice! I've marked this task as done: \n  ";
+        out += tasks.get(getIndex(input));
+        textWarper(out);
+    }
+
+    /**
+     * Prints out confirmations that the Task was added to the TaskList.
+     * 
+     * @param tasks The TaskList to be added to.
+     * @param temp  The Task to be added.
```

Comment:
Is there an extra space between 'temp' and 'The'?
Additionally, is there a reason to name it 'temp' instead of 'task'?